File: frameworks/domain.core/src/test/java/com/everest/frameworks/domain/core/DomainExceptionTest.java
package com.everest.frameworks.domain.core;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class DomainExceptionTest {

    @Test
    void should_be_able_to_construct_from_message() {
        DomainException exception = new DomainException("a message");
        assertThat(exception).isInstanceOf(RuntimeException.class);
        assertThat(exception).hasMessage("a message");
    }

    @Test
    void should_be_able_to_construct_from_message_and_cause() {
        IllegalArgumentException cause = new IllegalArgumentException();
        DomainException exception = new DomainException("a message", cause);
        assertThat(exception).hasMessage("a message");
        assertThat(exception).hasCause(cause);
    }

}

File: frameworks/domain.core/src/test/kotlin/com/qomolangma/frameworks/domain/core/DomainExceptionTest.kt
package com.qomolangma.frameworks.domain.core

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class DomainExceptionTest {
    @Test
    fun should_be_able_to_construct_from_message() {
        val exception = DomainException("a message")
        assertThat(exception).isInstanceOf(RuntimeException::class.java)
        assertThat(exception).hasMessage("a message")
    }

    @Test
    fun should_be_able_to_construct_from_message_and_cause() {
        val cause = IllegalArgumentException()
        val exception = DomainException("a message", cause)
        assertThat(exception).hasMessage("a message")
        assertThat(exception).hasCause(cause)
    }
}
